File: scheduler/core/src/main/java/io/orkes/conductor/scheduler/model/WorkflowScheduleModel.java
package io.orkes.conductor.scheduler.model;

import org.springframework.beans.BeanUtils;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
public class WorkflowScheduleModel extends WorkflowSchedule {

    public static WorkflowScheduleModel from(WorkflowSchedule schedule) {
        WorkflowScheduleModel model = new WorkflowScheduleModel();
        BeanUtils.copyProperties(schedule, model);
        return model;
    }

    @JsonIgnore
    public String getQueueMsgId() {
        return getName();
    }
}

File: scheduler/core/src/main/typescript/io/orkes/conductor/scheduler/model/WorkflowScheduleModel.ts
import { WorkflowSchedule } from './WorkflowSchedule';

export class WorkflowScheduleModel extends WorkflowSchedule {
    static from(schedule: WorkflowSchedule): WorkflowScheduleModel {
        const model = new WorkflowScheduleModel();
        Object.assign(model, schedule);
        return model;
    }

    getQueueMsgId(): string | undefined {
        return this.name;
    }
}
